src/eos-metrics-instrumentation.c

```diff
@@ -805,6 +815,89 @@ login_dbus_proxy_new (void)
     return dbus_proxy;
 }
 
+#define LOCATION_CONF_FILE SYSCONFDIR "/metrics/location.conf"
+#define LOCATION_LABEL_GROUP "Label"
+
+static gboolean
+record_location_label (gpointer unused)
+{
+    g_autoptr (GError) err = NULL;
+    g_autoptr (GKeyFile) kf = g_key_file_new ();
+
+    if (!g_key_file_load_from_file (kf, LOCATION_CONF_FILE, G_KEY_FILE_NONE, &err))
+      {
+        /* this file’s existence is optional, so not found is not an error */
+        if (g_error_matches (err, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_NOT_FOUND))
+          return G_SOURCE_REMOVE;
+
+        g_warning ("failed to load " LOCATION_CONF_FILE ", unable to record location label: %s",
+                   err->message);
+        return G_SOURCE_REMOVE;
+      }
+
+    g_auto (GStrv) keys = g_key_file_get_keys (kf, LOCATION_LABEL_GROUP, NULL, NULL);
+    if (keys == NULL || *keys == NULL)
+      return G_SOURCE_REMOVE;
+
+    g_autoptr (GString) label = g_string_new ("");
+    g_auto (GVariantBuilder) builder = { 0, };
```

Comment:
`G_VARIANT_BUILDER_INIT` exists for this purpose, although it requires GLib 2.50 (I don’t know what your current dependency is).